### PR TITLE
[WIP] Make max image upload size configurable

### PR DESCRIPTION
**Pull request by Flaburgan**
4194304 bytes is actually hardcoded everywhere. It's becoming too small in 2016, I wanted to make that customizable.

This is not working actually, I've got "gon is undefined", is there something special I should do to make gon available?

### CODE REVIEW

File: app/assets/javascripts/app/views/publisher/uploader_view.js [hunk 1]
```diff
@@ -7,7 +7,7 @@
 app.views.PublisherUploader = Backbone.View.extend({
 
   allowedExtensions: ["jpg", "jpeg", "png", "gif", "tif", "tiff"],
-  sizeLimit: 4194304,  // bytes
+  sizeLimit: gon.appConfig.settings.max_upload_size,
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/publisher/uploader_view.js]: Identifier 'max_upload_size' is not in camel case.

File: app/assets/javascripts/mobile/mobile_file_uploader.js [hunk 2]
```diff
@@ -11,7 +11,7 @@ function createUploader(){
        action: "/photos",
        debug: false,
        button: document.getElementById('file-upload-publisher'),
-       sizeLimit: 4194304,
+       sizeLimit: gon.appConfig.settings.max_upload_size,
```
Comment (diaspora-code-review) [inline app/assets/javascripts/mobile/mobile_file_uploader.js]: Identifier 'max_upload_size' is not in camel case.

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -150,7 +150,10 @@ def current_user_redirect_path
   def gon_set_appconfig
     gon.push(appConfig: {
                chat:     {enabled: AppConfig.chat.enabled?},
-               settings: {podname: AppConfig.settings.pod_name},
+               settings: {
+                 podname: AppConfig.settings.pod_name,
```
Comment (diaspora-code-review) [inline app/controllers/application_controller.rb]: Align the elements of a hash literal if they span more than one line.

File: app/assets/javascripts/mobile/mobile_file_uploader.js (hunk 2 above)
Comment (Flaburgan) [inline app/assets/javascripts/mobile/mobile_file_uploader.js]: I kept the same name in JS than in the diaspora.yml file here, should I change that?

**Comment by denschub** [comment]
Sorry, but no.

We've discussed this in #3653 before. We are not able to simply increase the upload limit and we should absolutely not expose a config option for that. Increasing the max image size would require the podmins to change their nginx/apache configs to increase the maximal body size. That, in turn, will lead to people increasing their max body sizes to a gigabyte or something and totally blowing up their ram.

Instead of randomly increasing the body limit and exposing configs that may break stuff if the podmin does not change their webserver configs we should implemented chunked uploading, which is tracked in #4997.

**Comment by Flaburgan** [comment]
Up to you, maybe if we don't expose it we can at least use a constant instead of a hard value everywhere, and increase the default value.

**Comment by Flaburgan** [comment]
So what do you think? Can we use a unique variable instead of hardcoded values, and where should I put this variable if we don't want to expose it?

**Pull request closed without merging**